src/git.rs

```diff
@@ -132,17 +132,18 @@ impl version_control::System for Repo {
             .map(|s| s.to_string())
             .collect::<HashSet<_>>();
 
-        all_files
+        let filtered_files = all_files
             .difference(&deleted_working_tree_files)
-            // Git reports files relative to the root of git root directory, so retrieve
-            // that and prepend it to the file paths.
             .map(|f| format!("{}", self.root.join(f).display()))
-            .map(|f| {
-                AbsPath::try_from(&f).with_context(|| {
-                    format!("Failed to find file while gathering files to lint: {}", f)
-                })
+            .filter_map(|f| match AbsPath::try_from(&f) {
```

Comment:
Do you have an example of when this is necessary? Without context, it looks like we're just eating the error when a given path cannot be accessed, which is not always correct. 

For example, if there is some other IO error, or there is a bug in the file selection code, this stuff would no longer error.

Reply:
Let's discuss this offline as it may be related to internal